src/Client/Common.php

```diff
@@ -43,4 +47,91 @@ protected function getUri(string $uri, bool $withCache): string
 
         return $uri;
     }
+
+    /**
+     * @param 'donation'|'mandate' $entityType
+     */
+    protected function postUpdateToSalesforce(string $uri, array $jsonSnapshot, string $uuid, string $entityType): Salesforce18Id
```

Comment:
I'm wondering if it would be slightly cleaner to pass the class name of the queued message verbatim, and use that along with static properties to derive `$uri` and `$entityType` here? Might be a little less error prone when editing and reduce the argument list size.

Reply:
Yes. I think it's not that error prone for now as Psalm checks that entityType is one of the two allowed values from the docblock. Deriving the uri based on the entity type seems like a good plan but I think we can do that as part of the change to use the `SALESFORCE_API_BASE` env var after its set in all environments.

At that point it might also be worth merging the clients that talk to SF  (Donation, Mandate, maybe also Campaign and Fund) into a single class as another simplification. So `\MatchBot\Client\Campaign::getById` would become `\MatchBot\Client\Salesforce:::getCampaignById` and `\MatchBot\Client\Mandate::createOrUpdate` would become `\MatchBot\Client\Salesforce::createOrUpdateMandate`.  Not sure about that part.